modules/web/src/com/haulmont/cuba/web/gui/components/WebTreeDataGrid.java

```diff
@@ -53,6 +53,22 @@ protected void onColumnReorder(Grid.ColumnReorderEvent e) {
         }
     }
 
+    @Override
+    public int getLevel(final Object itemId) {
+        int level = 0;
```

Comment:
Delegate `getLevel` to the wrapped component and add the `getLevel` method to `HierarchicalDataGridDataProvider`. Similar to Table delegation chain.